sympy/core/tests/test_sympify.py

```diff
@@ -615,3 +615,23 @@ def test_issue_13924():
     a = sympify(numpy.array([1]))
     assert isinstance(a, ImmutableDenseNDimArray)
     assert a[0] == 1
+
+
+def test_issue_14706():
```

Comment:
This should directly test sympify itself as well.

Reply:
What does that mean?

Reply:
I mean like `assert sympify(...) == ...`.

Reply:
I don't understand what you want me to test.  Is it related to this change?

Reply:
I mean to add tests like

```py
assert isinstance(sympify(np.array([1]), strict=True), np.array)
``` 

and so on. So that the new behavior is also tested directly.

Reply:
ah, okay.  To be honest, I don't even know what sympify does.  I assume it simplifies.  I'll add some tests tomorrow.  I'll probably need your help suggesting anything I'm missing.

Reply:
sympify converts its input to a SymPy type. So for instance `sympify(int(1))` creates `Integer(1)`. The strict=True option is used by operators like `+` (via `_sympify`), which disables things like converting strings.

Reply:
`assert isinstance(sympify(np.array([1]), strict=True), np.array)`

I don't understand what this is supposed to do, but it is failing.

Reply:
I guess it should be `np.ndarray`. But it also raises SympifyError, so it should be `raises(SympifyError, lambda: sympify(np.array([1]), strict=True))`.